cmd/thanos/bucket.go

```diff
@@ -85,10 +85,11 @@ func registerBucketVerify(m map[string]setupFunc, root *kingpin.CmdClause, name
 		var backupBkt objstore.Bucket
 		if len(backupconfContentYaml) == 0 {
 			if *repair {
-				return errors.Wrap(err, "repair is specified, so backup client is required")
+				return errors.New("repair is specified, so backup client is required")
 			}
 		} else {
-			backupBkt, err = client.NewBucket(logger, backupconfContentYaml, reg, name)
+			// nil Prometheus registerer: don't create conflicting metrics
```

Comment:
This is not a good solution. It's essentially as easy as `prometheus.WrapRegisterWithPrefix("backup_..., reg)` (:

Reply:
But also, not sure if it matters as it is only batch jobs, no one looks on metrics ;p

Reply:
Yeah, otherwise we register the same metrics twice.

Reply:
No, you don't: ` It's essentially as easy as prometheus.WrapRegisterWithPrefix("backup_..., reg) (:`